src/DependencyInjection/DoctrineExtension.php

```diff
@@ -625,6 +626,22 @@ protected function ormLoad(array $config, ContainerBuilder $container)
             $def
                 ->addTag('doctrine.event_listener', ['event' => Events::loadClassMetadata])
                 ->addTag('doctrine.event_listener', ['event' => Events::onClassMetadataNotFound]);
+
+            // Symfony 7.1 and higher expose type alias support in the EntityValueResolver
+            if (class_exists(EntityValueResolver::class)) {
+                $valueResolverReflection = new ReflectionClass(EntityValueResolver::class);
+
+                if ($valueResolverReflection->hasMethod('addTypeAlias')) {
```

Comment:
I think using a constructor argument instead of an adder can save us from using reflection at all. WDYT?

Reply:
Initially that is what I tried to do, but I figured explicitly checking support without depending on constructor argument count and order was more desirable. This also seems more in line with other parts of the extension which also call `add*` or `set*` methods. I do agree that reflection is generally undesirable.
Having this as a method also allows for easier extension of the ValueResolver beyond the bundle, since the property holding this mapping is otherwise private.

Also, setting the argument on its position by key doesn't seem to work. Replacing the reflection code in this PR with the following snippet:
```
$valueResolverDefinition->setArgument(3, $config['resolve_target_entities']);
```

Results in an error: `TypeError: Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver::__construct(): Argument #3 ($defaults) must be of type Symfony\Bridge\Doctrine\Attribute\MapEntity, array given`
Even when using 4 in the above code snippet. Using explicit argument names (`->setArgument('$typeAliases', ...)`) doesn't work as this causes a container compilation error on older bridge versions because the argument does not exist; this would break backwards compatibility.

Reply:
for add an argument, I think you can remove the condition ``if ($controllerResolverDefaults) {`` at l.527 for always declare the mapentity (with default arguments) as arg 2 and then you can decalre a argument 3.

Reply:
@NanoSector up to give @eltharin's suggestion a try?

Reply:
@nicolas-grekas Sorry, these PRs must have fallen off my radar. I've rebased them on the latest development branches and applied your and @eltharin 's suggestions.